pattern-lab/patternlab-node
# Feature: Add The Ability To Hide Viewall Links

## Summary
Adds the ability to hide any viewall links that would normally show up in the Pattern Lab navigation.

## Details
Updates the Webpack build config + `.patternlabrc` config to support a new `noViewAll` config option when customizing Pattern Lab's UI.

### Before vs After
![image](https://user-images.githubusercontent.com/1617209/62258309-dc7c8780-b3d7-11e9-8228-fbb70212ac90.png)

![image](https://user-images.githubusercontent.com/1617209/62258313-e1d9d200-b3d7-11e9-9406-9fb52e79132b.png)


## How To Test

1. Pull down this branch and install NPM dependencies by running `npm run setup`.
2. Go to the `packages/edition-twig` folder and uncomment line 5 in the example `.patternlabrc` config file.

```
module.exports = {
  buildDir: __dirname + '/public',
  wwwDir: __dirname + '/public/',
  publicPath: '/public/styleguide/',
  // noViewAll: true, // <!-- uncomment this line
};
```
3. Rebuild Pattern Lab's UI by running `npm run build:uikit` from the same `packages/edition-twig` folder
4. Confirm the regenerated UIKit is rendering properly by now running `npm run serve`

![image](https://user-images.githubusercontent.com/1617209/62258206-61b36c80-b3d7-11e9-9e1b-e853f1686e47.png)
